ext/openssl/ossl_pkey_ec.c

```diff
@@ -1021,6 +1021,100 @@ static VALUE ossl_ec_group_set_seed(VALUE self, VALUE seed)
 
 /* get/set curve GFp, GF2m */
 
+
+
+/*
+ * call-seq:
+ *   group.field_type   => Symbol
+ *
+ * See the OpenSSL documentation for EC_METHOD_get_field_type()
+ */
+static VALUE ossl_ec_group_get_field_type(VALUE self)
+{
+    EC_GROUP *group;
+    GetECGroup(self, group);
+    if (!group) {
+        rb_raise(eEC_GROUP, "EC_GROUP is not initialized");
+        return Qnil;
+    }
+
+    const EC_METHOD *method = EC_GROUP_method_of(group);
+    int field_type_id = EC_METHOD_get_field_type(method);
+    if (field_type_id == NID_X9_62_prime_field) {
+        return ID2SYM(s_GFp);
+    } else if (field_type_id == NID_X9_62_characteristic_two_field) {
+        return ID2SYM(s_GF2m);
+    }
+
+    return INT2NUM(field_type_id);
+}
+
+/*
+ * call-seq:
+ *   group.curve_params   => Array
+ *
+ * See the OpenSSL documentation for EC_GROUP_get_curve_GFp() and EC_GROUP_set_curve_GF2m()
+ */
+static VALUE ossl_ec_group_get_curve_params(VALUE self)
+{
+    EC_GROUP *group;
+    VALUE pBN, aBN, bBN;
+    BIGNUM *p, *a, *b;
+
+    GetECGroup(self, group);
+    if (!group) {
+        rb_raise(eEC_GROUP, "EC_GROUP not initialized");
+        return Qnil;
```

Comment:
`rb_raise()` does longjmp, so the return statement is unreachable and should be removed.